httpie/output/formatters/colors.py

```diff
@@ -78,13 +80,25 @@ def format_headers(self, headers: str) -> str:
 
     def format_body(self, body: str, mime: str) -> str:
         lexer = self.get_lexer_for_body(mime, body)
-        if lexer:
-            body = pygments.highlight(
-                code=body,
-                lexer=lexer,
-                formatter=self.formatter,
-            )
-        return body
+        if not lexer:
+            return body
+
+        # We may want to use a custom JSON lexer for special JSON data.
+        if type(lexer) is JsonLexer:
+            from ..utils import parse_prefixed_json
+            data_prefix = parse_prefixed_json(body)[0]
+            if data_prefix:
+                # `body` is prefixed with non-JSON data, likely a XSSI magic string.
+                # Use our own JSON lexer in order to properly display the data prefix
+                # and the actual JSON body.
+                from ..lexers.json import EnhancedJsonLexer
+                lexer = EnhancedJsonLexer()
```

Comment:
You no longer need to check for the prefix. Just swap it. The custom lexer is meant to handle both scenarios.

Reply:
And the could be moved to our `get_lexer()`. https://github.com/httpie/httpie/blob/464b5b4c1d8b3e72255739094e17a013d80abae7/httpie/output/formatters/colors.py#L154

Reply:
And now that there is a way to override the `Content-Type`, do you agree to remove that block?
https://github.com/httpie/httpie/blob/464b5b4c1d8b3e72255739094e17a013d80abae7/httpie/output/formatters/colors.py#L145-L152

Reply:
The idea behind the [explicit `--json` flag](https://httpie.io/docs#explicit-json) is to enable a maximally JSON-friendly mode. This extra detection is one of the features it enables (context: https://github.com/httpie/httpie/commit/74e4d0b67852ab0b628f687ac62e2d509fa74b61). 

With the new option for response content type overriding the user can achieve the same, but it takes more work and there’s a learning curve. 

The current behavior is not entirely elegant and I’m not sure how many people actually rely on it, but let’s leave it there at least for now, also for compatibility reasons.